test/test_channel.lua

```diff
@@ -0,0 +1,402 @@
+local test = require 'regress'
+local ipc = require 'libipc'
+
+test {
+   -- it should be possible to open a channel, write something to it
+   -- and read it back within the same thread.
+   openReadWriteSameThread = function()
+      local c = ipc.channel()
+      local data = 10
+      local status = c:write(data)
+      test.mustBeTrue(status == ":open")
```

Comment:
Suggestion: Expose the status strings as fields on channel, to allow something like: status == ipc.channel.OPEN, instead of hardcoding these strings everywhere.